service/service.go

```diff
@@ -666,14 +667,42 @@ func (s *Service) TestInstance(ctx context.Context, serverType ServerType, addre
 			}
 			return roleResponse.Role, roleResponse.Mode, resp.StatusCode, nil
 		}
+		makeIsLeaderRequest := func() (bool, error) {
+			if serverType != ServerTypeResilientSingle {
+				return false, nil
+			}
+			addr := net.JoinHostPort(address, strconv.Itoa(port))
+			url := fmt.Sprintf("%s://%s/_api/database", scheme, addr)
+			req, err := http.NewRequest("GET", url, nil)
+			if err != nil {
+				return false, maskAny(err)
+			}
+			if err := addJwtHeader(req, s.jwtSecret); err != nil {
+				return false, maskAny(err)
+			}
+			resp, err := client.Do(req)
+			if err != nil {
+				return false, maskAny(err)
+			}
+			if resp.StatusCode == 200 {
+				return true, nil
+			}
+			if resp.StatusCode == 503 && resp.Header.Get("X-Arango-Endpoint") != "" {
+				return false, nil
+			}
+			return false, maskAny(fmt.Errorf("Invalid status %d", resp.StatusCode))
+		}
 
 		for i := 0; i < 300; i++ {
 			if version, statusCode, err := makeVersionRequest(); err == nil {
 				if role, mode, statusCode, err := makeRoleRequest(); err == nil {
-					instanceUp <- instanceUpInfo{
-						Version: version,
-						Role:    role,
-						Mode:    mode,
+					if isLeader, err := makeIsLeaderRequest(); err == nil {
```

Comment:
This has no else part in which the error is reported or logged. Is this intentional?

Reply:
Yes, it will just retry